hollow/src/main/java/com/netflix/hollow/api/consumer/HollowConsumer.java

```diff
@@ -884,7 +885,29 @@ public B withRefreshListeners(HollowConsumer.RefreshListener... refreshListeners
         public B withGeneratedAPIClass(Class<? extends HollowAPI> generatedAPIClass) {
             if (HollowAPI.class.equals(generatedAPIClass))
                 throw new IllegalArgumentException("must provide a code generated API class");
-            this.apiFactory = new HollowAPIFactory.ForGeneratedAPI<>(generatedAPIClass);
+            this.generatedAPIClass = generatedAPIClass;
+            return (B)this;
+        }
+
+        /**
+         * Enable POJO caching for the specified types.
+         *
+         * When there is a type with a low cardinality, we can instantiate and cache a POJO implementation for each
+         * ordinal, which can be used by consumers in tight inner loops.  The types specified here will be cached in
+         * this way. This can only be used in conjunction with {@link #withGeneratedAPIClass(Class)}.
+         *
+         * @param cachedTypes one or more (low cardinality) types to cache as POJOs
+         * @return this builder
+         *
+         * @see <a href="https://hollow.how/advanced-topics/#caching">https://hollow.how/advanced-topics/#caching</a>
+         */
+        public B withCachedTypes(String...cachedTypes) {
```

Comment:
I think we need to specify this more clearly with regards to the client API. `POJO` is more of a producer term. 

Jotting down some notes.

In this case we are referring to hollow records and the choice between using the `HollowObjectFactoryProvider` when caching is not enabled and the `HollowObjectCacheProvider` when caching is enabled.  Such choice is performed by the generated API class (i.e. see `HollowConsumer.Builder.withGeneratedAPIClass`) which is instantiated by the HollowConsumer with the cached types, if any.  

The generated API class will use one of the factories, `HollowObjectFactoryProvider` or `HollowObjectCacheProvider` (the latter effectively uses the former) to instantiate an instance of a `HollowRecord` for the type and cache it.  The concrete type of the `HollowRecord` instance will be the generated API class associated with the type.  

When an instance of a type is returned from the client API, given say lookup its ordinal or via a search, the factory will be used.  When caching is enabled the same `HollowRecord` instance of a type will be returned (even if the consumer refreshes).

Caching may be disabled by calling the `detachCaches` method on the generated API (although once disabled it cannot be re-enabled).

Reply:
I originally wrote this as extra arguments to `withGeneratedAPIClass`, e.g.:

```java
public CinderConsumerBuilder withGeneratedAPIClass(
    Class<? extends HollowAPI> generatedAPIClass,
    String...cachedTypes) { ... }
```

Would this be preferable, @PaulSandoz? It would reinforce that the two go hand-in-hand.